src/test/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,37 @@
+package stringcalculator;
+
+import stringcalculator.number.Number;
+import stringcalculator.number.ZeroNumber;
+import stringcalculator.operator.OperatorSelector;
+
+public class StringCalculator {
+
+    private static final int SINGLE_STRING_LENGTH = 1;
+
+    private final String source;
+
+    public StringCalculator(String source) {
+        this.source = source;
+    }
+
+    public int add() {
+        if (isNullOrEmpty(source)) {
+            return new ZeroNumber().getValue();
+        }
+
+        if (isSingleLength(source)) {
+            return new Number(source).getValue();
+        }
+
+        return new OperatorSelector().select(source).add();
```

Comment:
```suggestion
        return new OperatorSelector(source).add();
```
OperatorSelector 에서 사용되어질 Selector 를 미리 정해 놓았기에
조금 더 간결하게 표현이 가능할 것 같아요 :)

Reply:
말씀주신것처럼 더 간결하게도 가능하겠네요.
근데 그러면... 클래스이름이 바뀌거나 해야할 듯 싶어요(이미 선택된 ..SelectedOperator?.. 네이밍 ㅠㅠ). 

그리고 지금은 Select를 해주는 클래스이기에 select라는 메세지를 보내서 클라이언트 객체가  select를 필요로 하는 시점까지 지연이가능한데, 말씀주신대로 변경하게되면 위의 네이밍처럼 생성자에서 바로 선택을해야하기에 select지연을 못하는 측면이 있을듯해요. 
물론 지연이 필요없는 클래스처럼 보이긴하지만, 말씀주신대로 변경해보니 부생성자에서 너무많은일을 해주는건 아닐까라는 생각이 들어서요.

리뷰어님 의견도 듣고싶습니다 :)